chromium_src/components/permissions/permission_context_base.cc

```diff
@@ -44,4 +59,75 @@ void PermissionContextBase::PermissionDecided(
       content_setting, is_one_time);
 }
 
+void PermissionContextBase::DecidePermission(
+    content::WebContents* web_contents,
+    const PermissionRequestID& id,
+    const GURL& requesting_origin,
+    const GURL& embedding_origin,
+    bool user_gesture,
+    BrowserPermissionCallback callback) {
+  PermissionContextBase_ChromiumImpl::DecidePermission(
+      web_contents, id, requesting_origin, embedding_origin, user_gesture,
+      std::move(callback));
+
+  if (content_settings_type() != ContentSettingsType::BRAVE_ETHEREUM)
```

Comment:
similar to https://github.com/brave/brave-core/pull/9443/files#r678622283, maybe we should have a method `IsGroupedPermissionType` that encapsulates this logic?

Reply:
https://github.com/brave/brave-core/pull/9443/commits/bfed18d3f531d2bbc98d69c7dd6a8bcc652cf0d3